File: ROS_Exercises/1-Exercise.1/C++/exercise1_number_counter.cpp
#include <ros/ros.h>
#include <std_msgs/Int64.h>

int counter = 0;
ros::Publisher pub;

void callback_number(const std_msgs::Int64& msg)
{
	ROS_INFO("Message received : %ld", msg.data);
	counter += 1;
	ROS_INFO("Counter : %d", counter);
	std_msgs::Int64 new_msg;
	new_msg.data = counter;
	pub.publish(new_msg);
}

int main (int argc, char **argv)
{
	ros::init(argc, argv, "number_counter");
	ros::NodeHandle nh;
	ros::Subscriber sub = nh.subscribe("/number", 1000, callback_number);
	pub = nh.advertise<std_msgs::Int64>("/number_count", 10);
	ros::spin();
}

File: ROS_Exercises/1-Exercise.1/Python/exercise1_number_counter.py
#!/usr/bin/env python

import rospy
from std_msgs.msg import Int64

counter = 0

def callback_number(msg):
    rospy.loginfo("Message received : ")
    rospy.loginfo(msg)
    global counter
    counter += 1
    rospy.loginfo("Counter : ")
    rospy.loginfo(counter)
    msg_new = Int64()
    msg_new.data=counter
    pub.publish(msg_new)

if __name__ == '__main__':
    rospy.init_node('number_counter')
    sub=rospy.Subscriber("/number", Int64, callback_number)
    pub = rospy.Publisher("/number_count", Int64, queue_size=10)
    rospy.spin()
